mods/other/tsm_chests/init.lua

```diff
@@ -58,10 +58,21 @@ minetest.register_node("tsm_chests:chest", {
 		local inv = meta:get_inventory()
 		inv:set_size("main", 8*4)
 	end,
-	can_dig = function(pos,player)
+	drop = "",
+	on_destruct = function(pos)
 		local meta = minetest.get_meta(pos);
 		local inv = meta:get_inventory()
-		return inv:is_empty("main")
+		-- iterate over chest contents, drop all
+		for i = 1, inv:get_size("main") do
+			local stack = inv:get_stack("main", i)
+			if not minetest.is_yes(minetest.settings:get("remove_items")) then
+				local obj = minetest.add_item(pos, stack:take_item(stack:get_count()))
+				if obj then
+					obj:set_velocity({x = math.random(-1, 1), y = 5, z = math.random(-1, 1)})
+				end
+			end
+			inv:set_stack("main", i, nil)
```

Comment:
```suggestion
			inv:set_stack("main", i)
```

nil is the value given when an argument isn't passed